src/components/sliding-panel.tsx

```diff
@@ -61,9 +65,51 @@ const SlidingPanel: FC<
 }) => {
   const node = useRef<HTMLDivElement>(null);
 
-  const onCloseRef = useRef(onClose);
+  const onCloseRef = useRef<(() => void) | null>(onClose);
```

Comment:
I've seen `useRef` for prop callbacks in a few places - I can't remember what the benefit of doing this is? Is it performance?

Reply:
Because those are functions that the user of the component will define without necessarily having knowledge of the implementation of the component. Ideally it would be a memoized function that gets passed, but that doesn't have to be the case, it could be an inlined function for example.
So while in some other places we do that for performance, here it's actually critical because we have 2 useEffects that depend on the fact that they're run either once only, or only on path changes, and they both use that onClose function, so if for whatever reason onClose get re-created in uniprot-website (because of a rerender of an unrelated prop) it would trigger those useEffects incorrectly.

Reply:
Regarding the suggested change, at the moment TS deduces the type from the component props, so I don't think that is necessary to specify it again here, and in any case, we don't have the case `| null` because onClose is required

Reply:
We're making the assumption then that `onClose` won't change after mounting the component. Which makes sense - I'm not sure when a consumer would want to change the `onClose` callback when it's already open.

Reply:
Yeah, I don't think it would happen either, but it is a caveat with this approach: "when did the user actually really wanted to change this?"